yksgi/src/lib.rs

```diff
@@ -0,0 +1,344 @@
+use llvm_sys::core::*;
+use llvm_sys::{LLVMOpcode, LLVMTypeKind};
+use std::collections::HashMap;
+use std::ffi::CStr;
+use std::ptr;
+
+mod llvmbridge;
+use llvmbridge::{LLVMModule, LLVMBasicBlock, LLVMValue, LLVMType, get_aot_original};
+
+/// Stopgap interpreter values.
+#[derive(Clone, Copy, PartialEq)]
+pub struct SGValue {
+    pub val: u64,
+    pub ty: LLVMType,
+}
+
+impl SGValue {
+    pub fn new(val: u64, ty: LLVMType) -> Self {
+        SGValue { val, ty }
+    }
+
+    pub fn with_type(&self, ty: LLVMType) -> Self {
+        SGValue::new(self.val, ty)
+    }
+
+    pub fn as_str(&self) -> String {
+        format!("SGValue ( {}, {:?} )", self.val, unsafe { self.ty.as_str() })
+    }
+}
+
+/// A frame holding live variables.
+struct Frame {
+    vars: HashMap<LLVMValue, SGValue>,
```

Comment:
Later we could consider using a FxHashMap (assuming we value performance over security).